kubernetes/structure_endpoint.go

```diff
@@ -0,0 +1,131 @@
+package kubernetes
+
+import (
+	api "k8s.io/api/core/v1"
+)
+
+func expandEndpointAddresses(in []interface{}) []api.EndpointAddress {
+	if len(in) == 0 {
+		return []api.EndpointAddress{}
+	}
+	addresses := make([]api.EndpointAddress, len(in))
+	for i, addr := range in {
+		r := api.EndpointAddress{}
+		addrCfg := addr.(map[string]interface{})
+
+		if v, ok := addrCfg["hostname"].(string); ok {
+			r.Hostname = v
+		}
+		if v, ok := addrCfg["ip"].(string); ok {
+			r.IP = v
+		}
+		if v, ok := addrCfg["node_name"].(string); ok {
+			if v != "" {
+				r.NodeName = ptrToString(v)
+			}
+		}
+		addresses[i] = r
+	}
+	return addresses
+}
+
+func expandEndpointPorts(in []interface{}) []api.EndpointPort {
+	if len(in) == 0 {
+		return []api.EndpointPort{}
+	}
+	subsets := make([]api.EndpointPort, len(in))
+	for i, port := range in {
+		r := api.EndpointPort{}
+		portCfg := port.(map[string]interface{})
+		if v, ok := portCfg["name"].(string); ok {
+			r.Name = v
+		}
+		if v, ok := portCfg["port"].(int); ok {
```

Comment:
Could you please use the same int type in both places here.

Reply:
This seems to be required because Terraform interprets the value as `int` but K8S struct wants an `int32`.  Am I missing an alternative?

Reply:
That makes sense. All good then.